bbq/compiler/compiler.go

```diff
@@ -3229,9 +3312,34 @@ func (c *Compiler[_, _]) compileEnumCaseDeclaration(
 	return
 }
 
-func (c *Compiler[_, _]) VisitAttachmentDeclaration(_ *ast.AttachmentDeclaration) (_ struct{}) {
-	// TODO
-	panic(errors.NewUnreachableError())
+func (c *Compiler[_, _]) VisitAttachmentDeclaration(declaration *ast.AttachmentDeclaration) (_ struct{}) {
+	// Similar to VisitCompositeDeclaration
+	// Not combined because need to access fields not accessible in CompositeLikeDeclaration
+	compositeType := c.DesugaredElaboration.CompositeDeclarationType(declaration)
+
+	c.compositeTypeStack.push(compositeType)
+	defer func() {
+		c.compositeTypeStack.pop()
+	}()
```

Comment:
The call inside the immediately invoked invoked function is already a function call, so can be used in `defer` as-is:
```suggestion
	defer c.compositeTypeStack.pop()
```